x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/data/BytesRefArrayVector.java

```diff
@@ -30,6 +34,25 @@ final class BytesRefArrayVector extends AbstractVector implements BytesRefVector
         this.values = values;
     }
 
+    static BytesRefArrayVector readArrayVector(int positions, StreamInput in, BlockFactory blockFactory) throws IOException {
+        final BytesRefArray values = new BytesRefArray(in, blockFactory.bigArrays());
```

Comment:
This is the main improvement in this PR. The rest is for a follow-up where we will implement specialized serialization for ArrayBlock.